services/cache-headers.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const assert = require('assert')
+const Joi = require('joi')
+const coalesce = require('../lib/coalesce')
+
+const serverStartTimeGMTString = new Date().toGMTString()
+const serverStartTimestamp = Date.now()
+
+const queryParamSchema = Joi.object({
+  // Not using nonNegativeInteger because it's not required.
+  maxAge: Joi.number()
+    .integer()
+    .min(0),
+}).required()
+
+function coalesceCacheLength(
+  cacheHeaderConfig,
+  serviceCacheLengthSeconds,
+  queryParams
+) {
+  const { defaultCacheLengthSeconds } = cacheHeaderConfig
+  // The config returns a number so this would only fail if we break the
+  // wiring. Better to fail obviously than silently.
+  assert(defaultCacheLengthSeconds !== undefined)
+
+  const ourCacheLength = coalesce(
+    serviceCacheLengthSeconds,
+    defaultCacheLengthSeconds
+  )
+
+  const { value: { maxAge: overrideCacheLength } = {}, error } = Joi.validate(
```

Comment:
I find this hard to read. What is the meaning of the additional `= {}`?

Reply:
When there's an error, Joi will return something like `{ error: ... }`. You can't destructure undefined, which would result in `TypeError: Cannot destructure property `bar` of 'undefined' or 'null'`. The `= {}` is a default value for `value` that avoids the crash.

This is how destructuring works:

```jsconsole
> var { foo } = undefined
TypeError: Cannot destructure property `foo` of 'undefined' or 'null'.
> var { foo } = {}
undefined
> foo
undefined
```

And how a default value `{ foo } = {}` figures in:

```jsconsole
> var { bar: { foo } } = {}
TypeError: Cannot destructure property `foo` of 'undefined' or 'null'.
> var { bar: { foo } = {} } = {}
undefined
> foo
undefined
```